src/transformers/modeling_utils.py

```diff
@@ -1437,10 +1437,20 @@ def _resize_token_embeddings(self, new_num_tokens, pad_to_multiple_of=None):
             add_hook_to_module(new_embeddings, hook)
         self.set_input_embeddings(new_embeddings)
 
+        # Update new_num_tokens with the actual size of new_embeddings
+        if pad_to_multiple_of is not None:
+            if is_deepspeed_zero3_enabled():
+                import deepspeed
+
+                with deepspeed.zero.GatheredParameters(new_embeddings.weight, modifier_rank=None):
+                    new_num_tokens = new_embeddings.weight.shape[0]
+            else:
+                new_num_tokens = new_embeddings.weight.shape[0]
+
```

Comment:
@kai01ai, I think it's enough to delete this code block and pass `new_num_tokens` as it is in this function scope to `_get_resized_lm_head()`.
Because `new_embeddings` variable is created based on `new_num_tokens` on `_get_resized_embeddings()`.
So it doesn't need to be reassigned from `new_embeddings.weight.shape[0]`.

Is there any case whether `new_num_tokens` get changed?

Reply:
The `new_num_tokens` can be changed in `_get_resized_embeddings` because of the `pad_to_mulitple_of`, which is why we can't do this. However if we can use new_embedding.num_embeddings, with both deepspeed and not deepspeed then sure! Would you like to open a pr for this? 

Totally fine for you to review like this, it's the preferred method for code discussions!